File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s1, String s2) {
    if (s1.length() != s2.length())
			return false;
		HashMap<Character, Character> map = new HashMap<>();
		HashMap<Character, Character> map1 = new HashMap<>();
		for (int i = 0; i<s1.length(); i++) {
			char ch = s1.charAt(i);
			char ch1 = s2.charAt(i);
			if (map.containsKey(ch)) {
				if (map.get(ch) != ch1) {
					return false;
				}
			}
			if (map1.containsKey(ch1)) {
				if (map1.get(ch1) != ch) {
					return false;
				}
			}
			map.put(ch, ch1);
			map1.put(ch1, ch);
		}
		return true;    
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
           return False
        mp = {}
        mp1 = {}
        i = 0
        while i < len(s):
              a = s[i]
              b = t[i]
              if a in mp and mp.get(a) != b:
                 return False
              if b in mp1 and mp1.get(b) != a:
                 return False
              
              mp[a] = b
              mp1[b] = a
              i += 1
        
        return True